lib/matplotlib/axes/_base.py

```diff
@@ -2436,14 +2436,25 @@ def autoscale_view(self, tight=None, scalex=True, scaley=True):
         if tight is not None:
             self._tight = bool(tight)
 
-        if self.use_sticky_edges and (
-                (self._xmargin and scalex and self._autoscaleXon) or
-                (self._ymargin and scaley and self._autoscaleYon)):
-            stickies = [artist.sticky_edges for artist in self.get_children()]
-        else:  # Small optimization.
-            stickies = []
-        x_stickies = np.sort([x for sticky in stickies for x in sticky.x])
-        y_stickies = np.sort([y for sticky in stickies for y in sticky.y])
+        x_stickies = y_stickies = np.array([])
+        if self.use_sticky_edges:
+            # Only iterate over axes and artists if needed.  The check for
+            # ``hasattr(ax, "lines")`` is necessary because this can be called
+            # very early in the axes init process (e.g., for twin axes) when
+            # these attributes don't even exist yet, in which case
+            # `get_children` would raise an AttributeError.
+            if self._xmargin and scalex and self._autoscaleXon:
+                x_stickies = np.sort(np.concatenate([
+                    artist.sticky_edges.x
```

Comment:
Even w/ the comment, the `hasattr(ax, "lines")` is a bit mysterious to me.  Why are you singling out `lines` in particular versus other children?  Why does `ax.get_children` error out?  It should just return None, shouldn't it?

Reply:
It doesn't matter which children you pick, the point is that for twinned axes this is called even before the lines attribute is *created* (i.e. the first time `self.lines = []` is called on the Axes ever), so get_children() just raises an AttributeError when it tries to return `self.lines` (among other children).  (Yes, Axes init is a mess.)

Reply:
I believe you, but this seems to be a band-aid rather than fixing the underlying issue, which is that `get_children` is not working or `self.lines` should be set (I guess by calling `cla`)?  

I don't understand how `self.lines` is not created since `__init__` calls `cla`...

Reply:
Yes, but cla() calls `self._sharex.get_xlim()` even before setting `self.lines`, and get_xlim causes a call to unstale_viewLim and tragedy occurs.
Really self.lines should be set even earlier in `__init__`, and cla() should not do `self.lines = []` but just `for child in self.get_children(): child.remove()` (IOW it's not cla()'s job to create the attributes, just to clear them out) but I'm not going to touch that here...

Reply:
Can we change `get_children` to do:
```
if not hasattr(self, 'lines'):
    # sometimes we haven't even initialized the artist lists yet:
   return []    # or None?  

```

Reply:
Actually I think this is more likely to hide bugs -- normally no one should call get_children() that early in the axes init process, so I'd rather just workaround this in the sole place I know this can happen.  But I can also apply your patch if you really prefer that.

Reply:
Its just really mystifying why the check would be there, even with the comment.

Reply:
It is a shorthand for "is the Axes actually built yet?".  Agree it would be better if we could be sure that we never call this method before the Axes is fully set up, but short of re-writing the Axes init process (which is if I understand it one of our biggest performance bottle knecks) and making sure things like this that need to be computed don't get computed until as late as possible, we have to put a band-aid someplace.

I agree with @anntzer that putting this check in `get_children` is likely to mask other bugs in addition to solving this one.

We can't put the fix outside of this function because we may need the rest of it to run.

We could explicitly add state `_done_initing`, but that get awkward when you start to have sub-classes as now the order (or if!) the subclass calls super ends up really mattering.